### PR TITLE
Feature/ds4 30 agv move to charging station

### PR DESCRIPTION
**Pull request by 3FCO**
This makes it so that we can't send the AGV to charge at a 100% battery, as it will just get stuck forever...
Also removed a dependency AGVAPIClient in the application pom.xml as it's not needed

### CODE REVIEW

File: backend/AGV/src/main/java/dk/sdu/controllab/agv/service/AGVServiceImpl.java [hunk 1]
```diff
@@ -30,23 +30,27 @@ public AGVStatusResponse getStatus() {
...
 
-    private boolean canLoadNewProgram() {
-        return getStatus().getState().equals(AGVState.IDLE);
+    private boolean canLoadNewProgram(AGVProgram program) {
+        if (getStatus().getState().equals(AGVState.EXECUTE) ||
+                getStatus().getBattery() == 100 && program.equals(AGVProgram.MOVE_TO_CHARGER)) {
+            return false;
+        }
+
+        return true;
```
Comment (MagnusHJensen) [inline backend/AGV/src/main/java/dk/sdu/controllab/agv/service/AGVServiceImpl.java:48]: Maybe we should start utilizing custom runtime exceptions?
Then we add a RestControllerAdvice, which handles any custom errors thrown and maps it to a response body.

See [RecipeErrorHandler](https://github.com/SDU-F23/SDU-F23-Semester-Project/blob/main/backend/Recipe/src/main/java/dk/sdu/controllab/recipe/controller/RecipeErrorHandler.java) for inspiration.

Comment (3FCO) [reply backend/AGV/src/main/java/dk/sdu/controllab/agv/service/AGVServiceImpl.java:48]: I need some more context on this. I couldn't find any source looking like your example, or how to actually use it for that matter

Comment (MagnusHJensen) [reply backend/AGV/src/main/java/dk/sdu/controllab/agv/service/AGVServiceImpl.java:48]: Maybe this makes sense, https://medium.com/@Seonggil/handling-exceptions-with-restcontrolleradvice-exceptionhandler-e7c95216da8d

But it looks good!

File: backend/AGV/src/main/java/dk/sdu/controllab/agv/service/AGVServiceImpl.java [hunk 2]
```diff
@@ -30,23 +30,27 @@ public AGVStatusResponse getStatus() {
...
         loadProgram(program);
         executeProgram();
     }
 
 
-    private boolean canLoadNewProgram() {
-        return getStatus().getState().equals(AGVState.IDLE);
+    private boolean canLoadNewProgram(AGVProgram program) {
+        if (getStatus().getState().equals(AGVState.EXECUTE) ||
+                getStatus().getBattery() == 100 && program.equals(AGVProgram.MOVE_TO_CHARGER)) {
```
Comment (MagnusHJensen) [inline backend/AGV/src/main/java/dk/sdu/controllab/agv/service/AGVServiceImpl.java:44]:
```suggestion
                (getStatus().getBattery() == 100 && program.equals(AGVProgram.MOVE_TO_CHARGER))) {
```

**Review by MagnusHJensen — APPROVED** [review]

**Pull request merged**